lib/sharing/ProvideSharedPlugin.js

```diff
@@ -41,29 +44,39 @@ class ProvideSharedPlugin {
 	 */
 	constructor(options) {
 		validate(options);
-
-		this._provides = /** @type {[string, ProvideOptions][]} */ (
-			parseOptions(
-				options.provides,
-				item => {
-					if (Array.isArray(item))
-						throw new Error("Unexpected array of provides");
-					/** @type {ProvideOptions} */
-					const result = {
-						shareKey: item,
-						version: undefined,
-						shareScope: options.shareScope || "default",
-						eager: false
-					};
-					return result;
-				},
-				item => ({
-					shareKey: item.shareKey,
-					version: item.version,
-					shareScope: item.shareScope || options.shareScope || "default",
-					eager: !!item.eager
-				})
-			)
+		/** @type {[string, ProvideOptions][]} */
+		this._provides = parseOptions(
```

Comment:
Don't do it, you break types

Reply:
Apologies, I did not notice this when I merged `main` branch into my local branch. Will fix it